script.js

```diff
@@ -0,0 +1,356 @@
+      // scroll to center
+      return (this.viewWidth - listWidth) / 2 + this.margin;
+    } else if (nodeLeft + nodeWidth >
+                          -this.translateX + this.viewWidth + this.margin) {
+      // scroll left
+      newTranslate = this.viewWidth - nodeLeft - nodeWidth + this.margin;
+    } else if (nodeLeft < -this.translateX + this.margin) {
+      // scroll right
+      newTranslate = -nodeLeft + this.margin;
+    }
+
+    // If the new scroll offset contains first/last node, we have to align the
+    // list to begin/end.
+    if (this.lastNode.offsetLeft + this.lastNode.offsetWidth <=
+                          -newTranslate + this.viewWidth + this.margin) {
+      return this.viewWidth + this.margin -
+                          this.lastNode.offsetLeft - this.lastNode.offsetWidth;
+    } else if (this.firstNode.offsetLeft >= -newTranslate + this.margin) {
+      return -this.firstNode.offsetLeft + this.margin;
+    }
+    return newTranslate;
+  };
+
+  /**
+   * Handle focus from spacial navigator
+   */
+  proto._handleFocus = function(itemElem) {
+    var nodeElem = this._getNodeFromItem(itemElem);
+    this._scrollTo(itemElem);
+    this._fireEvent('focus', {
+      itemElem: itemElem,
+      nodeElem: nodeElem,
+      index: Array.prototype.indexOf.call(this.children, nodeElem)
+    });
+
+    // Since we may have many elements in one node, node-changed event happens
+    // when active node is changed.
+    if (this.currentNode && this.currentNode !== nodeElem) {
+      this.currentNode.classList.remove('active');
+      this._fireEvent('node-changed', {
+        scrollbox: this,
+        oldNodeElem: this.currentNode,
+        newNodeElem: nodeElem
+      });
+    }
+    this.currentNode = nodeElem;
+    this.currentNode.classList.add('active');
+    itemElem.focus();
+  };
+
+  /**
+   * Handle unfocus from spacial navigator
+   */
+  proto._handleUnfocus = function(itemElem) {
+    var nodeElem = this._getNodeFromItem(itemElem);
+    this._fireEvent('unfocus', {
+      itemElem: itemElem,
+      nodeElem: nodeElem,
+      index: Array.prototype.indexOf.call(this.children, nodeElem)
+    });
+    itemElem.blur();
+  };
+
+  /**
+   * Add/Remove elements to/from spacial navigator when a child is
+   * added/removed.
+   */
+  proto._handleMutation = function(mutations) {
+    mutations.forEach(function(mutation) {
+      var prevElem = mutation.previousSibling;
+      var nextElem = mutation.nextSibling;
+      if (prevElem) {
+        prevElem = (prevElem.nodeType === Node.ELEMENT_NODE)?
+                                    prevElem : prevElem.previousElementSibling;
+      }
+      if (nextElem) {
+        nextElem = (nextElem.nodeType === Node.ELEMENT_NODE)?
+                                    nextElem : nextElem.nextElementSibling;
+      }
+      var navigableElems;
+      var nodes;
+      if (mutation.type === 'childList') {
+        nodes = Array.prototype.slice.call(mutation.addedNodes);
```

Comment:
I would suggest to separate the code about addedNodes and removedNodes as two different functions. That makes the code more readable.